src/server/origin/cph-number/controller.js

```diff
@@ -29,32 +31,41 @@ export const getController = {
  */
 export const postController = {
   handler(req, res) {
-    const { cphNumber } = /** @type {CphNumberPayload} */ (req.payload)
+    const { cphNumber, nextPage } = /** @type {CphNumberPayload & NextPage} */ (
+      req.payload
+    )
     // Remove whitespace from cphNumber
     const input = cphNumber ? cphNumber.replace(/\s+/g, '') : cphNumber
-    const [isValid, message] = validator(input)
+    const { isValid, errors } = validator(input)
 
     if (!isValid) {
-      req.yar.clear('cphNumber')
+      req.yar.set('origin', {
+        ...req.yar.get('origin'),
+        cphNumber: undefined
+      })
+
       return res.view(indexView, {
+        nextPage,
```

Comment:
I've added nextPage here as otherwise after an error the redirect was not happening. I've also added a unit test for it. A similar change is needed in the other controllers after an error.